src/main/scala/ing/wbaa/druid/client/DruidAdvancedHttpClient.scala

```diff
@@ -415,20 +418,23 @@ object DruidAdvancedHttpClient extends DruidClientBuilder {
             )
           }
         }
-        .map {
+        .mapAsync(settings.maxOpenRequests) {
```

Comment:
@Fokko @mmartina @bjgbeelen @anskarl I need a bit of advice here. This seems to me the correct value for the max concurrency parameter of `mapAsync`, but that's more of an educated guess than an informed decision.

Reply:
I couldn't find any useful recommendation on this. Most examples use pretty low values though. And the default maxOpenRequests = 128 seems quite high.
From aka.http.scaladsl.HttpExt, line number 849
"// a connection pool can never have more than pipeliningLimit * maxConnections requests in flight at any point
    val parallelism = settings.pipeliningLimit * settings.maxConnections "
but in this case it is applied to .mapAsyncUnordered, not .mapAsync.
I think it is best to parameterize it with a low default value, for example druid-advanced-http-client.entity-parallelism = 4

Reply:
Improved in 2441f73478d8c7d5cd16c248536af72a2d0821ac